ng/src/web/pages/credentials/dialog.js

```diff
@@ -114,206 +118,225 @@ class CredentialsDialog extends React.Component {
-            name="privacy_algorithm"
-            onChange={onValueChange}/>
-          <Radio
-            value={SNMP_PRIVACY_ALGORITHM_DES}
-            title="DES"
-            checked={privacy_algorithm === SNMP_PRIVACY_ALGORITHM_DES}
-            name="privacy_algorithm"
-            onChange={onValueChange}/>
-          <Radio
-            value={SNMP_PRIVACY_ALOGRITHM_NONE}
-            title={_('None')}
-            checked={privacy_algorithm === SNMP_PRIVACY_ALOGRITHM_NONE}
-            name="privacy_algorithm"
-            onChange={onValueChange}/>
-        </FormGroup>
-      </Layout>
+      <SaveDialog
+        visible={visible}
+        title={title}
+        onClose={onClose}
+        onSave={onSave}
+        initialData={{...DEFAULTS, ...credential}}
+      >
+        {({
+          data: state,
+          onValueChange,
+        }) => {
+          return (
+            <Layout flex="column">
+
+              <FormGroup title={_('Name')}>
+                <TextField
+                  name="name"
+                  grow="1"
+                  value={state.name}
+                  size="30"
+                  onChange={onValueChange}
+                  maxLength="80"/>
+              </FormGroup>
+
+              <FormGroup title={_('Comment')}>
+                <TextField
+                  name="comment"
+                  grow="1"
+                  value={state.comment}
+                  size="30"
+                  maxLength="400"
+                  onChange={onValueChange}/>
+              </FormGroup>
+
+              <FormGroup title={_('Type')}>
+                <Select
+                  onChange={value => this.handleTypeChange(
+                    value, state.autogenerate, onValueChange)}
+                  disabled={is_edit}
+                  items={typeOptions}
+                  value={state.base}/>
+              </FormGroup>
+
+              <FormGroup title={_('Allow insecure use')}>
+                <YesNoRadio
+                  name="allow_insecure"
+                  value={state.allow_insecure}
+                  onChange={onValueChange}/>
+              </FormGroup>
+
+              <FormGroup
+                title={_('Auto-generate')}
+                condition={(state.base === USERNAME_PASSWORD_CREDENTIAL_TYPE ||
+                  state.base === USERNAME_SSH_KEY_CREDENTIAL_TYPE) && !is_edit}>
+                <YesNoRadio
+                  name="autogenerate"
+                  value={state.autogenerate}
+                  onChange={onValueChange}/>
+              </FormGroup>
+
+              <FormGroup
+                title={_('SNMP Community')}
+                condition={state.base === SNMP_CREDENTIAL_TYPE}>
+                {is_edit &&
+                  <Checkbox
+                    name="change_community"
+                    checked={state.change_community === '1'}
+                    checkedValue="1"
+                    unCheckedValue="0"
```

Comment:
```jsx
checkedValue={YES_VALUE}
unCheckedValue={NO_VALUE}
```
(Not sure if that isn't the default already in Checkbox component).

Reply:
Doesn't seem to be the case. I changed dialog.js to your suggestion for now.